### PR TITLE
Feature mes 1548 render empty slot

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 1548 - Render empty slot

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [x] PO's approval
emptyslot

### CODE REVIEW

File: src/pages/journal/components/empty-slot/empty-slot.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
+empty-slot {
+  ion-card {
+    &.grey-bg {
+      background: transparent;
+    }
+    border: 4px dashed #4A4A4A;
+    ion-grid {
```
Comment (johnmorleybjss) [inline src/pages/journal/components/empty-slot/empty-slot.scss]: Is there any way to remove the `ion-grid` style? I can't see it as part of your template, so I'm assuming it's part of the internals of one of the Ionic components.

File: src/pages/journal/components/empty-slot/empty-slot.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
+empty-slot {
+  ion-card {
+    &.grey-bg {
+      background: transparent;
+    }
+    border: 4px dashed #4A4A4A;
```
Comment (ammarhaiderbjss) [inline src/pages/journal/components/empty-slot/empty-slot.scss]: The design in abstract has the border width set at `2px`

File: src/pages/journal/components/empty-slot/empty-slot.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+  .spacing-column {
+    max-width: 152px;
+    display: flex;
+    justify-content: center;
+    .activity-code {
+      font-weight: bold;
+    }
+  }
+  .slot-changed-indicator {
```
Comment (johnmorleybjss) [inline src/pages/journal/components/empty-slot/empty-slot.scss]: Are these bits about slot change still relevant to an empty slot?

File: src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { Observable } from 'rxjs/Observable';
```
Comment (Cavalier07) [inline src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts]: We already have a mock at - src/pages/journal/components/test-slot/__mocks__/screen-orientation.mock.ts

Could we not reuse/improve that one

File: src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+export class ScreenOrientationMock  {
```
Comment (ammarhaiderbjss) [inline src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts]: Did we try this? https://github.com/chrisgriffith/ionic-native-mocks

File: src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts (hunk 4 above)
Comment (Cavalier07) [reply src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts]: or do what ammar said ;)

File: src/pages/journal/components/empty-slot/empty-slot.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  @Input()
+  hasSlotChanged: boolean;
+
+  @Input()
+  showLocation: boolean;
+
+  constructor(public screenOrientation: ScreenOrientation) {}
+
+  isPortrait() : boolean {
```
Comment (Cavalier07) [inline src/pages/journal/components/empty-slot/empty-slot.ts]: Can we move this up to SlotComponent and then remove it from TestSlotComponent

File: src/pages/journal/components/empty-slot/empty-slot.scss (hunk 2 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/empty-slot.scss]: Done

File: src/pages/journal/components/empty-slot/empty-slot.scss (hunk 1 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/empty-slot.scss]: I based my sccs on another - i've removed it and see no discernible difference.

File: src/pages/journal/components/empty-slot/empty-slot.scss (hunk 3 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/empty-slot.scss]: removed

File: src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts (hunk 5 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts]: the mocks folder and contents were duplicated by accident. have removed them

File: src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/__mocks__/screen-orientation.mock.ts]: I accidentally duplicated the screen orientation mock - have removed and using the original.

File: src/pages/journal/components/empty-slot/empty-slot.ts (hunk 6 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/empty-slot.ts]: SlotComponent is an interface not a class so can't move the method into the interface.
(I know in C# 7.0 (or later) they are allowing this, but did a quick check on typescript and this isn't supported.

File: src/pages/journal/components/empty-slot/empty-slot.scss [hunk 7]
```diff
@@ -0,0 +1,36 @@
+empty-slot {
+  ion-card {
+    &.grey-bg {
```
Comment (ammarhaiderbjss) [inline src/pages/journal/components/empty-slot/empty-slot.scss]: realise I suggested this name, but its not really a grey bg! can we re-name to transparent-bg?

File: src/pages/journal/components/empty-slot/empty-slot.ts (hunk 6 above)
Comment (Cavalier07) [reply src/pages/journal/components/empty-slot/empty-slot.ts]: so many languages with so many different rules . Happy to close this one

File: src/pages/journal/components/empty-slot/empty-slot.scss (hunk 7 above)
Comment (rossfauldsbjss) [reply src/pages/journal/components/empty-slot/empty-slot.scss]: I agree - transparent-bg is a much better name will change now.

**Pull request merged**